hooks/templatedatavalidator.js

```diff
@@ -0,0 +1,111 @@
+var fs = require('fs');
+var path = require('path');
+const { parseJamboConfig } = require('../commands/helpers/utils/jamboconfigutils');
+const { error } = require('../commands/helpers/utils/logger');
+/**
+ * Validates page template's data configuration
+ * 
+ * @param {Object} pageData data that gets passed into a page template
+ * @returns {boolean} false if validator should throw an error
+ */
+module.exports = function (pageData) {
+  const jamboConfig = parseJamboConfig();
+  const validatorResults = [
+    isGlobalConfigValid(pageData.global_config), 
+    isPageVerticalConfigValid(pageData, jamboConfig)
+  ];
+  const isValid = !validatorResults.some((result) => !result);
+  return isValid;
+}
+
+/**
+ * Validates global config for the page template
+ * 
+ * @param {Object} globalConfig 
+ * @returns {boolean}
+ */
+function isGlobalConfigValid(globalConfig) {
+  if (!globalConfig.experienceKey) {
+    error('Missing Info: no experienceKey found.');
+    return false;
+  }
+  return true;
+}
+
+/**
+ * Valivates vertical config for current page template
+ * 
+ * @param {Object} pageData 
+ * @param {Object} jamboConfig 
+ * @returns {boolean}
+ */
+function isPageVerticalConfigValid(pageData, jamboConfig) {
+  const themeDirectory = path.join(jamboConfig.dirs.themes, jamboConfig.defaultTheme);
+  return Object.keys(pageData.verticalsToConfig).reduce((isValid, key) => {
```

Comment:
Ah shoot ok the isValid is actually kind of tricky too. (Sorry I know I'm really harping on this PR lol...) Maybe we could use `.map` and then call `.every` instead of just `.reduce` ?

Reply:
no problem. This is a good idea!